src/random/linear_number_generator.cpp

```diff
@@ -0,0 +1,13 @@
+#include "linear_number_generator.h"
+
+LinearNumberGenerator::LinearNumberGenerator(int base_seed)
+                                            : RandomNumberGenerator(base_seed), modulus((1 << 16) + 1), multiplier(75), increment(74) {};
+
+LinearNumberGenerator::LinearNumberGenerator(int base_seed, int modulus, int multiplier, int increment)
+                                            : RandomNumberGenerator(base_seed), modulus(modulus), multiplier(multiplier), increment(increment) {};
+

```

Comment:
My suggestion to reduce long parameter list and organize code:
in LinearNumberGenerator.h
`struct LinearNumberGeneratorParameters{
    int modulus = (1 << 16) + 1
   int multiplier = 75;
   int increment = 74;
};`

in LinearNumberGenerator.cpp instead of selection
`
LinearNumberGenerator::LinearNumberGenerator(int seed, LinearNumberGeneratorParameters params={.modulus = (1<<16) + 1, .multiplier = 75, .increment=74 }) : RandomNumberGenerator(seed), params(params) {}
` params(params) and use params struct inside generator class, or copy each param individually - up to you